pkg/cmd/step/syntax/step_syntax_effective.go

```diff
@@ -451,6 +466,26 @@ func (o *StepSyntaxEffectiveOptions) createPipelineForKind(kind string, lifecycl
 	return parsed, nil
 }
 
+// RemoveDuplicateEnvVars removes any duplicate env var definitions
+func RemoveDuplicateEnvVars(env []corev1.EnvVar) []corev1.EnvVar {
+	counts := map[string]int{}
+	for _, e := range env {
+		counts[e.Name] = counts[e.Name] + 1
+	}
+	for name, c := range counts {
```

Comment:
We don't care which env variable is left in the end as long as there are no duplicated?

Reply:
fwiw the existing test cases were generating duplicate name + values which is why I added this code. I could always just leave the duplicates there if you'd rather?